channels/drive/client/drive_file.c

```diff
@@ -122,7 +122,10 @@ static BOOL drive_file_remove_dir(const WCHAR* path)
 	}
 
 	if (!path)
```

Comment:
Well, this is kinda bad. Note that `path` is referenced on the line 115. The compiler is likely to remove this condition entirely on the basis that NULL check is redundant because passing NULL to _wcslen() is undefined behavior thus it must never be NULL.

I'd move this check to the start of the function, before doing anything, and drop the `free()` call.